tests/testthat/test-parser.R

```diff
@@ -70,3 +70,37 @@ test_that("Calling parse_gitlog with incorrect git repo path returns correct err
   })
 })
 
+
+# A repo with 3 commits, the first adds hello.R, the second renames it to hi.R, the last adds bye.R
+test_that("Git repo with 3 commits, first file is hello.R, then renamed to hi.R, then adds another file bye.R", {
+  # Create a temporary directory for the Git repository
+  tools_path <- file.path(tools_path)
+  tool <- yaml::read_yaml(tools_path)
+  perceval_path <- tool[["perceval"]]
+  git_repo_path <- suppressWarnings(example_repo_three_commits())
+  result <- parse_gitlog(perceval_path, git_repo_path)
+  # expect only 1 because prefix test files and suffix example files are ignored.
+  expect_equal(nrow(result), 3)
+  suppressWarnings(git_delete_sample_log(git_repo_path))
```

Comment:
You should not be suppressing warnings. If they are appearing, it is because something is wrong and you should fix it. Indeed, `git_delete_sample_log` is not deleting any folder here. Because of that, the folders don't get deleted, and the next time you run the unit test, it leads to strange errors.